src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs

```diff
@@ -77,10 +98,15 @@ internal override void Authenticate([NotNull]ReplicatorOptionsDictionary options
             var authDict = new AuthOptionsDictionary
             {
                 Username = Username,
-                Password = Password,
                 Type = AuthType.HttpBasic
             };
 
+            // TODO string Password will be deprecated and replaced with byte array password
+            if (String.IsNullOrEmpty(Password))
+                authDict.Password = new NetworkCredential(string.Empty, PasswordSecureString).Password;
```

Comment:
What is the goal of this?  This defeats the purpose of using `SecureString` by changing it back to a string.  Is it not possible to serialize a secure string?

Reply:
I am getting `System.ArgumentException : Cannot encode System.Security.SecureString to Fleece!`